### ISSUE TITLE
Should getPortalUrl() take IHubRequestOptions as well as IPortal?

### ISSUE THREAD

**Issue by tomwayson** [Esri/hub.js#319]
We could move the following logic down into `getPortalUrl()` :

https://github.com/Esri/hub.js/blob/9f01ab16ba29b40e67d9e274ef8cbad86e47e0b0/packages/common/src/resources/get-item-thumbnail-url.ts#L26-L31

That way the same logic could be applied to other places that call `getPortalUrl()` like `getPortalHomeUrl()` and `getPortalApiUrl()`, `getPortalDataUrl()`, etc.

**Comment by tomwayson** [comment]
Actually, maybe the logic should be:


```js
// this is IRequestOptions.portal, see: https://esri.github.io/arcgis-rest-js/api/request/IRequestOptions/#portal
const portalApiUrl = portalOrRequestOptions.portal;
if (typeof portalApiUrl === 'string') {
  return portalApiUrl.replace(/\/sharing\/rest\/?$/, '');
}
const portal = portalOrRequestOptions.portalSelf 
   ? // assume request options was passed in
     portalOrRequestOptions.portalSelf 
   : // assume a portal was passed in
     portalOrRequestOptions;
// rest of function stays the same
```

**Comment by tomwayson** [comment]
this was resolved in #333
